src/components/accounts/ledger/SignInLedger.js

```diff
@@ -6,14 +6,27 @@ import FormButton from '../../common/FormButton';
 import { Translate } from 'react-localize-redux';
 import LedgerConfirmActionModal from './LedgerConfirmActionModal';
 import { signInWithLedger, clear, redirectToApp, refreshAccount } from '../../../actions/account';
+import RequestStatusBox from '../../common/RequestStatusBox'
 
 export function SignInLedger(props) {
     const dispatch = useDispatch();
     const account = useSelector(({ account }) => account);
-    const signingIn = account.actionsPending.includes('SIGN_IN_WITH_LEDGER')
 
-    const handleSignIn = async () => {
+    const gettingAccounts = account.actionsPending.includes('GET_LEDGER_ACCOUNT_IDS')
+    const addingAccounts = account.actionsPending.includes('ADD_LEDGER_ACCOUNT_ID')
+    const savingAccounts = account.actionsPending.includes('SAVE_AND_SELECT_LEDGER_ACCOUNTS')
+    const signingIn = gettingAccounts || addingAccounts || savingAccounts
+
+    const ledgerAccounts = addingAccounts && Object.keys(account.signInWithLedger).map((accountId) => ({
+        accountId,
+        status: account.signInWithLedger[accountId] === 'waiting'
```

Comment:
not sure about this structure. Isn't it easier to just always have status in `signInWithLedger[accountId].status` and avoid having to use `typeof` and this nested ternary?

Reply:
Good point. Done.